### ISSUE TITLE
how to get log output during development of wizard pages?

### ISSUE THREAD

**Issue by hgy59** [SynoCommunity/spksrc#6224]
When developing wizard pages, I can write log output to a log file from within the wizard shell script files.

But I can't figure out where I can find the output of `{ console.log(...); }` called by java script code.

Another solution, instead of using `console.log`, would be to write to a log file from java script. Any suggestions on how to write to a (server local) file from java script code of a package installation wizard?

**Comment by smaarn** [comment]
```
But I can't figure out where I can find the output of { console.log(...); } called by java script code.
```

From what I recall this goes into your actual web browser logs.

**Comment by mreid-tt** [comment]
@hgy59 this would usually be in your browser's developer console... see: https://javascript.info/devtools for some further information. Happy to collaborate on this if you are trying to resolve a particular issue as I've gone down the rabbit hole a few times when developing the wizards for packages like ownCloud and a few others.

**Comment by hgy59** [comment]
@mreid-tt I want to check the version of existing minio data format in existing data shares in the upgrade wizard of minio (#5555).
I think it is best to get this information in the activate function of a wizard page (or deactivate of the previous page).

Since you have used this kind of functions in the wizards of the owncloud package, you probably have some hints for me?

**Comment by hgy59** [comment]
> ```
> But I can't figure out where I can find the output of { console.log(...); } called by java script code.
> ```
> 
> From what I recall this goes into your actual web browser logs.

@smaarn thank you!  Yes, it is there 😛

**Comment by hgy59** [comment]
So I see that java script is on the client, but I need to make checks on the server when navigating to a wizard page...

**Comment by mreid-tt** [comment]
Well you won't be able to dynamically get details on the fly however you can run some specific code and store the results in variables that can then be used by JavaScript. See the checks that I helped with in SABnzbd: https://github.com/Safihre/spksrc/blob/master/spk/sabnzbd/src/wizard/upgrade_uifile.sh

**Comment by hgy59** [comment]
I think in the wizard I can't make checks on the server and have to postpone checks to a validate_* installer function.

@mreid-tt compared with the owncloud install wizard: 
The wizard does not validate whether the provided restore path exists on the server.
This validation is done in `service_postinst` (to be honest, it is not validated, but the restore is skipped when the backup file is not found or has unexpected name).

**Comment by hgy59** [comment]
> Well you won't be able to dynamically get details on the fly however you can run some specific code and store the results in variables that can then be used by JavaScript. See the checks that I helped with in SABnzbd: https://github.com/Safihre/spksrc/blob/master/spk/sabnzbd/src/wizard/upgrade_uifile.sh

Ok, The minio data format can be evaluated for an existing share path when starting the upgrade wizard.
This can be used to display the existing format on a wizard page.
But when the shared folder is modified in the upgrade wizard, it is not possible to make a roundtrip to the server to check whether this folder exists, contains a minio data structure, and so on.
Those checks must be post poned to the installation code.

**Comment by hgy59** [comment]
finally solved with 99cd8f49e83edb91c52870b20407bdc68d02839e
